ansible_base/authentication/views/authenticator_users.py

```diff
@@ -0,0 +1,40 @@
+import importlib
+import logging
+from types import ModuleType
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+
+from ansible_base.lib.utils.views.django_app_api import AnsibleBaseDjangoAppApiView
+
+logger = logging.getLogger('ansible_base.authentication.views.authenticator_users')
+
+
+class BadViewSet(AnsibleBaseDjangoAppApiView):
+    pass
+
+
+def get_parent_view():
+    try:
+        user_viewset_name = settings.ANSIBLE_BASE_USER_VIEWSET
+        module_name, junk, class_name = user_viewset_name.rpartition('.')
+        module = importlib.import_module(module_name, package=class_name)
+        user_viewset_view = getattr(module, class_name)
+        if isinstance(user_viewset_view, ModuleType):
+            raise ModuleNotFoundError()
+        return user_viewset_view
+    except ModuleNotFoundError:
+        logger.error("ANSIBLE_BASE_USER_VIEWSET was not an APIView")
+    except AttributeError:
+        logger.debug("ANSIBLE_BASE_USER_VIEWSET was not specified")
+
+    return BadViewSet()
+
+
+class AuthenticatorPluginRelatedUsersView(get_parent_view()):
+    def get_queryset(self, **kwargs):
+        authenticator_id = kwargs.get('authenticator_id', None)
+        if authenticator_id is None:
+            return get_user_model().objects.filter(pk=-1)
```

Comment:
Maybe `get_user_model().objects.none()` here to avoid a round trip to the db server. It should do the same thing (return an empty queryset).

Reply:
I had initially tried `none()` first but I was getting errors but they now seem to be working. Code updated.